config/src/config/mod.rs

```diff
@@ -177,52 +177,58 @@ impl NodeConfig {
 
     pub fn set_data_dir(&mut self, path: PathBuf) -> Result<()> {
         self.base = Arc::new(BaseConfig::new(path, self.base.role));
-        self.prepare()?;
+        self.prepare();
         Ok(())
     }
 
     pub fn set_role(&mut self, role: RoleType) -> Result<()> {
         self.base = Arc::new(BaseConfig::new(self.base.data_dir.clone(), role));
-        self.prepare()?;
+        self.prepare();
```

Comment:
btw not sure I understand the point of calling prepare every time, and if there is a point, what if someone forgets to call it?